AnyType/Project_2/AnyType.cpp

```diff
@@ -0,0 +1,160 @@
+#include"DataStructures.h"
+#include"AnyType.h"
+#include<iostream>
+
+
+
+AnyType::AnyType()// empty constructor
+{
+	ValueType = Types::UNKNOWN_TYPE;
+}
+
+AnyType::~AnyType() = default;
+
+AnyType::AnyType(const AnyType &obj)//copy constructor
+{
+	data = obj.data;
+	ValueType = obj.ValueType;
+}
+
+AnyType::AnyType(AnyType &&obj)//move constructor
+{
+	data = obj.data;
+	ValueType = obj.ValueType;
+}
+
+char AnyType::ToChar()
+{
+	if (ValueType != Types::CHAR)// check if current value type != char
+		throw AnyType::BadCastException();//if it's not char throw bad cast exception
+	return data.charData;//if it's char - return char value
+}
+
+int AnyType::ToInt()
+{
+	if (ValueType != Types::INT)
+		throw AnyType::BadCastException();
+	return data.intData;
+}
+
+double AnyType::ToDouble()
+{
+	if (ValueType != Types::DOUBLE)
+		throw AnyType::BadCastException();
+	return data.doubleData;
+}
+
+bool AnyType::ToBool()
+{
+	if (ValueType != Types::BOOL)
+		throw AnyType::BadCastException();
+	return data.boolData;
+}
+
+float AnyType::ToFloat()
+{
+	if (ValueType != Types::FLOAT)
+		throw AnyType::BadCastException();
+	return (data.floatData);
+}
+
+void AnyType::operator=(AnyType const & obj)//copy assigment
+{
+	if (this != &obj)
+	{
+		data = obj.data;
+		ValueType = obj.ValueType;
+	}
+}
+
+void AnyType::operator=(AnyType &&obj)//move assigment
+{
+	if (this != &obj)
+	{
+		data = obj.data;
+		ValueType = obj.ValueType;
+	}
+}
+
+void AnyType::operator=(bool var)//overloaded operator =(we get the data type(in this case bool) and fill field of this type in the union)
+{
+	data.boolData = var;
+	ValueType = Types::BOOL;
+}
+//we'll do the same below for all types
+
+void AnyType::operator=(int var)
+{
+	data.intData = var;
+	ValueType = Types::INT;
+}
+
+void AnyType::operator=(float var)
+{
+	data.floatData = var;
+	ValueType = Types::FLOAT;
+}
+
+void AnyType::operator=(double var)
+{
+	data.doubleData = var;
+	ValueType = Types::DOUBLE;
+}
+
+std::ostream& operator<<(std::ostream &out, const AnyType &anyType)//overloaded << for print // get data and type mhetod
+{
```

Comment:
I would suggest that client should be provided with an opportunity to receive object type as a string/char* or enum id